apis/kueue/v1alpha1/workload_webhook_test.go

```diff
@@ -0,0 +1,80 @@
+package v1alpha1
+
+import (
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// makeWorkload is copied from "sigs.k8s.io/kueue/pkg/util/testing"
+// NOTE: we could not import "sigs.k8s.io/kueue/pkg/util/testing" because of cyclic import.
```

Comment:
I think you can avoid the circular dependency doing this: https://github.com/golang/go/wiki/CodeReviewComments#import-dot

Reply:
@alculquicondor Hi, I have switched to dot import, but it seems a workaround to me, should me move the validation logic to a separate package, such as `pkg/validation`?

Reply:
yeah, that's actually a better idea.

Let's do it in `apis/kueue/v1alpha1/validation`

Reply:
hmm I just realize this won't work, the circular dependency still persists: the package `apis/kueue/v1alpha1/validation` imports the `Worload` struct from `apis/kueue/v1alpha1`, and `apis/kueue/v1alpha1` imports `apis/kueue/v1alpha1/validation` for the validator. The root cause is that the validator function is called inside `apis/kueue/v1alpha1` package.

Reply:
`Workload` doesn't have to implement the `Defaulter` or `Validator` interface. You could create a new type for it that lives in a webhooks package.

Reply:
Regarding of the definition of `Defaulter` or `Validator` interface:
```go
type Defaulter interface {
	runtime.Object
	Default()
}
type Validator interface {
	runtime.Object
	ValidateCreate() error
	ValidateUpdate(old runtime.Object) error
	ValidateDelete() error
}
```
it seems these interfaces could only be implements on the `Workload` struct, otherwise we could not access the fields.

Reply:
ahhh, I missed that they depend on `runtime.Object` :disappointed:

I guess we have to go the dotimport route then.